src/t_stream.c

```diff
@@ -1493,6 +1626,11 @@ size_t streamReplyWithRange(client *c, stream *s, streamID *start, streamID *end
     while(streamIteratorGetID(&si,&id,&numfields)) {
         /* Update the group last_id if needed. */
         if (group && streamCompareID(&id,&group->last_id) > 0) {
+            if (group->offset && streamIsContiguousRange(s,&id)) {
+                group->offset++;
+            } else if (s->offset) {
+                group->offset = streamGetOffset(s,&id);
```

Comment:
please add a comment that explains what we do here.
also, this call includes an iteration (raxSeek), are we sure we can afford the extra work / complexity here? isn't that a nested loop?

Reply:
Added a comment.

WRT to extra work, although the iterator is used in the worst-case, it is still a valid concern. I'm not sure how to respond short of trying to benchmark a little. On the other hand, this looks like an argument in favor of keeping 'first_id' as part of the stream struct.